drivers/wifi/nrf700x/zephyr/src/zephyr_net_if.c

```diff
@@ -196,6 +196,13 @@ static void ip_maddr_event_handler(struct net_mgmt_event_callback *cb,
 		return;
 	}
 
+	if (mgmt_event != NET_EVENT_IPV4_MADDR_ADD &&
+	    mgmt_event != NET_EVENT_IPV4_MADDR_DEL &&
+	    mgmt_event != NET_EVENT_IPV6_MADDR_ADD &&
+	    mgmt_event != NET_EVENT_IPV6_MADDR_DEL) {
+		return;
+	}
```

Comment:
I'd put this even before the call to `wifi_nrf_get_vif_ctx` and `net_if_get_device calls since it will save you doing those lookups for spurious events